test/clt-tests/sharding/replication/replication-test.rec

```diff
@@ -0,0 +1,282 @@
+––– input –––
+export INSTANCE=2
+––– output –––
+––– block: ../base/flexible/start-searchd-precach –––
+––– input –––
+export INSTANCE=3
+––– output –––
+––– block: ../base/flexible/start-searchd-precach –––
+––– input –––
+export INSTANCE=4
+––– output –––
+––– block: ../base/flexible/start-searchd-precach –––
+––– input –––
+export INSTANCE=5
+––– output –––
+––– block: ../base/flexible/start-searchd-precach –––
+––– input –––
+export CLUSTER_NAME=replication
+––– output –––
+––– block: ../base/flexible/create-cluster –––
+––– input –––
+export CLUSTER_NAME=replication
+––– output –––
+––– block: ../base/flexible/join-cluster-on-all-nodes –––
+––– input –––
+mysql -h0 -P1306 -e "CREATE TABLE testrt (title text, content text, gid uint);" && mysql -h0 -P1306 -e "INSERT INTO testrt VALUES(0,'List of HP business laptops','Elitebook Probook',1);" && mysql -h0 -P1306 -e "INSERT INTO testrt VALUES(6775836565403860994,'List of Dell business laptops','Latitude Precision Vostro',2);" && mysql -h0 -P1306 -e "ALTER CLUSTER replication ADD testrt;" && mysql -h0 -P1306 -e "SHOW STATUS LIKE 'cluster_replication_indexes';"
+––– output –––
++-----------------------------+--------+
+| Counter                     | Value  |
++-----------------------------+--------+
+| cluster_replication_indexes | testrt |
++-----------------------------+--------+
+––– input –––
+mysql -h0 -P1306 -e "SELECT * FROM testrt\G;"
+––– output –––
+*************************** 1. row ***************************
+id: 6775836565403860994
+title: List of Dell business laptops
+content: Latitude Precision Vostro
+gid: 2
+*************************** 2. row ***************************
+id: %{NUMBER}
+title: List of HP business laptops
+content: Elitebook Probook
+gid: 1
+––– input –––
+export INSTANCE=5
+––– output –––
+––– block: ../base/flexible/stop-searchd –––
+––– input –––
+mysql -h0 -P1306 -e "INSERT INTO replication:testrt VALUES(0,'List of HP business laptops','Macbook',3);"; echo $?
+––– output –––
+0
+––– input –––
+export INSTANCE=5
+––– output –––
+––– block: ../base/flexible/start-searchd-precach –––
+––– input –––
+sleep 5; mysql -h0 -P5306 -e "SELECT * FROM testrt\G;"
+––– output –––
+*************************** 1. row ***************************
+id: %{NUMBER}
+title: List of Dell business laptops
+content: Latitude Precision Vostro
+gid: 2
+*************************** 2. row ***************************
+id: %{NUMBER}
+title: List of HP business laptops
+content: Macbook
+gid: 3
+*************************** 3. row ***************************
+id: %{NUMBER}
+title: List of HP business laptops
+content: Elitebook Probook
+gid: 1
+––– input –––
+export INSTANCE=2
+––– output –––
+––– block: ../base/flexible/stop-searchd –––
+––– input –––
+mysql -h0 -P3306 -e "INSERT INTO replication:testrt VALUES(0,'NEW','TABLE',4);"; echo $?
+––– output –––
+0
+––– block: ../base/flexible/start-searchd-precach –––
+––– input –––
+sleep 5;mysql -h0 -P2306 -e "SELECT * FROM testrt\G;"
```

Comment:
It looks like we're testing multiple outputs from different nodes and expecting them to match. It would be more efficient to handle this with a single command using a loop, for instance, and validate the outputs using temporary files.

Here's what I suggest: For each node, save the output to a file, run a diff to check for discrepancies between nodes, and confirm that the data matches what was previously inserted by validating only one time.